README.md

```diff
@@ -68,34 +68,49 @@ import dame_flame
 df = pd.read_csv("dame_flame/data/sample.csv")
 result = dame_flame.DAME_FLAME.DAME(input_data=df, treatment_column_name="treated", outcome_column_name="outcome", holdout_data=1.0)
 print(result[0])
-#>    x1   x2   x3   x4
-#> 0   0   1    1    *
-#> 1   0   1    1    *
-#> 2   1   0    *    *
-#> 3   1   0    *    *
+#>    x1   x2   x3   x4    weights
+#> 0   0   1    1    *     1
+#> 1   0   1    1    *     1
+#> 2   1   0    *    1     1
+#> 3   1   0    *    1     1
+
+print(result[1])
+#> [[0,1],[2,3]]
 ```
-result is a list, where the first element in the list is of type **Data Frame**. The dataframe contains all of the units that were matched, and the covariates and corresponding values, that it was matched on. The covariates that each unit was not matched on is denoted with a " * " character. The list 'result' will have additional values based on additional optional parameters, detailed in additional documentation below. 
+result is a list, where the first element in the list is of type **Data Frame**. The dataframe contains all of the units that were matched, and the covariates and corresponding values, that it was matched on. The covariates that each unit was not matched on is denoted with a " * " character. This element also includes a column of unit weights which specifies the number of times each unit was matched. The second element is a list of unit IDs belonging to each of the matched groups created by the algorithm. The list 'result' will have additional values based on additional optional parameters, detailed in additional documentation below. 
```

Comment:
"the number of times each unit was matched" could be confusing bc does that mean the number of units I was matched to or the number of groups I'm in?

So maybe, "the number of main matched groups a unit is part of" or something? 

Can you detail the second element a little more? I think I said something about this below as well, in the grouped_mr.py comments. So it's a list of lists, where each list contains the unit ids belonging to a main matched group? Or is it any group beyond just main matched groups -- do clarify this.

Reply:
Also I'm realizing that hopefully this isn't confusing: "The list 'result' will have additional values based on additional optional parameters, detailed in additional documentation below."

Would you be able to check in the documentation for where we mention the additional optional parameters and make sure we don't call them the "1st" and "2nd" or something like that, because now the ordinality would be changed.

I guess if we wanted we could make your list of matched group ids also an optional parameter to return and even store/compute. If we did that though, we would need two versions of the find_mmg_of_unit function, etc, so it's maybe not worth the hassle? I recall you saying you thought adding these new data structs wasn't really a change to the time complexity in your speed trials, but if it is and it's worth it, go ahead with it I guess?

Reply:
I do not see any mention of the ordinality of the optional parameters, so I think we're good on that front!

As for making the second element of the list 'result' optional, that is a very interesting idea. I think it would be worth it if we discover that computing MGs significantly affects the algorithm's speed on big datasets. If we do make it optional, we already have your mmg_of_unit and te_of_unit functions which are equipped to recover that information without the additional return. We would just need to update the mmg function to work when repeats = False. Also I think it would be a good idea to merge our mmg and te functions into two functions (as opposed to four) which can determine which method to use on their own. Other than that, the ATE function would need to be modified to be able to recover a complete list of MGs post-matching, but I do believe that this is possible (albeit slow).

Reply:
RE: "I think it would be worth it if we discover that computing MGs significantly affects the algorithm's speed on big datasets"

If I remember correctly, you said you didn't think it made a speed decrease -- do you remember what the size you ran it on for that was?

Reply:
For 5000 units with 20 covariates the modified version was about 2 seconds faster and took about 87 seconds. For 10000 units with 30 covariates, the modified version was about 3 seconds slower and took about 175 seconds. I haven't tested with any larger datasets though.